L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -49,21 +49,18 @@ void HGCalConcentratorSuperTriggerCellImpl::createAllTriggerCells(
       }
 
       l1t::HGCalTriggerCell triggerCell;
-
-      GlobalPoint point;
-      if (fixedDataSizePerHGCROC_ == true && thickness > kHighDensityThickness_) {
-        point = coarseTCmapping_.getCoarseTriggerCellPosition(coarseTCmapping_.getCoarseTriggerCellId(id));
-      } else {
-        point = triggerTools_.getTCPosition(id);
-      }
-      math::PtEtaPhiMLorentzVector p4(0, point.eta(), point.phi(), 0.);
-      triggerCell.setPosition(point);
-      triggerCell.setP4(p4);
       triggerCell.setDetId(id);
       if (energyDivisionType_ == superTriggerCell && id != s.second.getMaxId()) {
         continue;
       }
-      trigCellVecOutput.push_back(triggerCell);
+
+      //To guard against the case in v8 geometry where 
+      //there might be different thicknesses within a module 
+      if ( superTCmapping_.getCoarseTriggerCellId(id)!=s.second.getSTCId() ){
```

Comment:
One more thing: since it is not supposed to happen in the V9 geometry, it would be good to add an exception if this happens with a V9 geometry.

Reply:
They are only removed if the thickness of the given cell within the superTC does not correspond to the thickness of the superTC (the detID of which is one of the constituent cells).

Reply:
I believe that you have checked that there is no loss of energy due to this.

Reply:
This is a very small effect. In principle I think there can be energy loss, if the stcMaxId TC happens to be a different thickness to the stcId. I could try to think of some guard against this.

Reply:
Ok. Let's keep it that way for the moment. It's for V8 samples only and we shouldn't use V8 samples anymore, except maybe for some specific cases.
But write in the comments that this can happen.